DOCS/CASMNET-1534 - Improve linking of CSM install and upgrade to information mgmt. net and bi-can docs (#1671)
Cray-HPE/docs-csm: upgrade/1.2/scripts/sls/sls_updater.py_technical_details.md

> # `sls_updater.py` Technical Details

Reviewer: I see that this page was based on the contents taken from the SLS Upgrade README file -- should this file be linked from that one?

Author: good catch.  added linking between the two pages.

> **No Action needed.  Informational Purposes only.**

Reviewer: ```suggestion
**No action needed. Informational purposes only.**
```

> ## Actions and Order

Reviewer: ```suggestion
## Actions and order
```

This migration is performed offline for data security. The running SLS file is first dumped, then the
migration script is run and a new, migrated output file is created.

1. Migrate switch naming (in order): `leaf` to `leaf-bmc`, and `agg` to `leaf`.
1. Remove `api-gateway` entries from HMLB subnets for CSM 1.2 security.
1. Remove `kubeapi-vip` reservations for all networks except NMN.
1. Create the new BICAN "toggle" network.
1. Migrate the existing CAN to CMN.
1. Create the CHN network.
1. Convert IP addresses of the CAN network.
1. Create MetalLB Pools and ASN entries on CMN and NMN networks.
1. Update `uai_macvlan` in NMN DHCP ranges and `uai_macvlan` VLAN.
1. Remove unused user networks (CAN or CHN) if requested (`--retain-unused-user-network` to keep).

> ## Migrate Switch Names

Reviewer: ```suggestion
## Migrate switch names
```

Switch names change in CSM 1.2 and must be applied in the following order:

1. `leaf` switches become `leaf-bmc` switches.
1. `agg` switches become `leaf` switches.

This needs to be done in the order listed above.

> ## Remove `api-gateway` / `istio-ingress-gateway` Reservations from HMNLB Subnets

Reviewer: ```suggestion
## Remove `api-gateway` / `istio-ingress-gateway` reservations from HMNLB subnets
```

> For CSM 1.2, the API gateway will no longer listen on the HMNLB MetalLB address pool.

Reviewer: ```suggestion
For CSM 1.2, the API gateway no longer listens on the HMNLB MetalLB address pool.
```

> These aliases provided DNS records and are being removed.

Reviewer: ```suggestion
These aliases provided DNS records and have been removed in CSM 1.2.
```

> ## Create the BICAN Network "Toggle"

Reviewer: ```suggestion
## Create the BICAN network "toggle"
```

> New for CSM 1.2, the BICAN network `ExtraProperties` value of `SystemDefaultRoute` is used

Reviewer: ```suggestion
New for CSM 1.2: The BICAN network `ExtraProperties` value of `SystemDefaultRoute` is used
```

to point to the CAN, CHN, or CMN, and is used by utilities to systematically toggle routes.

> ## Migrate (existing) CAN to (new) CMN

Reviewer: ```suggestion
## Migrate existing CAN to new CMN
```

> Using the existing CAN as a template, create the CMN.  The same IP addresses will be preserved for

Reviewer: ```suggestion
Using the existing CAN as a template, create the CMN. The same IP addresses will be preserved for
```

> NCNs (`bootstrap_dhcp`).  A new `network_hardware` subnet will be created where the end of the previous

Reviewer: ```suggestion
NCNs (`bootstrap_dhcp`). A new `network_hardware` subnet will be created where the end of the previous
```

`bootstrap_dhcp` subnet existed to contain switching hardware. MetalLB pools in the `bootstrap_dhcp`
subnet will be shifted around to remain at the end of the new bootstrap subnet.

> ## Create the CHN Network

Reviewer: ```suggestion
## Create the CHN
```

> With the original CAN as a template, the new CHN network will be created. IP addresses will come from the

Reviewer: ```suggestion
With the original CAN as a template, the new CHN will be created. IP addresses will come from the
```

`--customer-highspeed-network <vlan> <ipaddress>` (or its defaults). This is be created by default, but
can be removed (if not needed or desired) by using the `--retain-unused-user-network` flag.

> ## Convert the IP Addresses of the CAN Network

Reviewer: ```suggestion
## Convert the IP addresses of the CAN
```

Since the original/existing CAN has been converted to the new CMN, the CAN must have new IP addresses.
These are provided using the `--customer-access-network <vlan> <ipaddress>` (or its defaults). This CAN
conversion will happen by default, but the new CAN may be removed (if not needed or desired) by using the
`--retain-unused-user-network` flag.

> ## Add BGP Peering Information to CMN and NMN

Reviewer: ```suggestion
## Add BGP peering information to CMN and NMN
```

MetalLB and switches now obtain BGP peers using SLS data.

```text
  --bgp-asn INTEGER RANGE         The autonomous system number for BGP router
                                  [default: 65533;64512<=x<=65534]
  --bgp-cmn-asn INTEGER RANGE     The autonomous system number for CMN BGP
                                  clients  [default: 65534;64512<=x<=65534]
  --bgp-nmn-asn INTEGER RANGE     The autonomous system number for NMN BGP
                                  clients  [default: 65533;64512<=x<=65534]
```

In CMN and NMN:

```yaml
  "Type": "ethernet",
  "ExtraProperties": {
    "CIDR": "10.102.3.0/25",
    "MTU": 9000,
    "MyASN": 65536,
    "PeerASN": 65533,
    "Subnets": 
```

> ## Remove `kubeapi-vip` Reservations For All Networks Except NMN

Reviewer: ```suggestion
## Remove `kubeapi-vip` reservations for all networks except NMN
```

Self explanatory. This endpoint now exists only on the NMN.

> ## Update `uai_macvlan` in NMN Ranges and `uai_macvlan` VLAN

Reviewer: ```suggestion
## Update `uai_macvlan` in NMN ranges and `uai_macvlan` VLAN
```

Self explanatory. Ranges are used for the addresses of UAIs.

> ## Remove Unused User Networks (Either CAN or CHN) if Desired

Reviewer: ```suggestion
## Remove unused user networks (either CAN or CHN) if desired
```

> By default the CAN will be removed if `--bican-user-network-name CHN` or the CHN will be removed if

Reviewer: ```suggestion
By default, the CAN will be removed if `--bican-user-network-name CHN` is specified, or the CHN will be removed if
```

> `--bican-user-network-name CAN`. In order to keep this network, use the `--retain-unused-user-network` flag.

Reviewer: ```suggestion
`--bican-user-network-name CAN` is specified. In order to keep a network from being removed, use the `--retain-unused-user-network` flag.
```
